gensim/models/tfidfmodel.py

```diff
@@ -78,10 +81,41 @@ def __init__(self, corpus=None, id2word=None, dictionary=None,
         and it will be used to directly construct the inverse document frequency
         mapping (then `corpus`, if specified, is ignored).
         """
-        self.normalize = normalize
         self.id2word = id2word
-        self.wlocal, self.wglobal = wlocal, wglobal
+        self.wlocal, self.wglobal, self.normalize = wlocal, wglobal, normalize
         self.num_docs, self.num_nnz, self.idfs = None, None, None
+        n_tf, n_df, n_n = smartirs
+        self.smartirs = smartirs
+
+        if self.wlocal is None:
+            if n_tf == "n":
+                self.wlocal = lambda tf, mean=None, _max=None: tf
```

Comment:
better to use simple define (instead of lambda) for avoiding pickle problems (here and everywhere)